core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -1800,9 +1800,12 @@ private void findSubQueries(
     case ALL:
       switch (logic) {
       case TRUE_FALSE_UNKNOWN:
-        if (validator.getValidatedNodeType(node).isNullable()) {
-          break;
-        } else if (true) {
```

Comment:
I am confused while looking at the original logic. Looks like the switch case would break no matter if type is nullable or not. What's the point of calling getValidateNodeType() here? Should we just remove these couple of lines all together?

Reply:
> What's the point of calling getValidateNodeType() here?

I believe calling getValidateNodeType() is to make sure that every node has validated type. If no, it will throw exception.

Reply:
But with your change, this check is actually removed. If it's safe to do so, why not just completely remove these lines?

Reply:
The difference is that with my change it returns directly instead of executing the following code` bb.registerSubQuery(node, logic);`.

Reply:
I see. thx.